src/reconcile/rec_write.c

```diff
@@ -1892,7 +1904,7 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
      * If re-instantiating this page in memory (either because eviction wants to, or because we
      * skipped updates to build the disk image), save a copy of the disk image.
      */
-    if (F_ISSET(r, WT_REC_SCRUB) || (r->cache_write_restore && multi->supd != NULL))
+    if (F_ISSET(r, WT_REC_SCRUB) || multi->supd_restore)
```

Comment:
I could have sworn I asked about this, but I must have somehow failed to post the question. Anyway...

Doesn't `WT_REC_SCRUB` now have to set `multi->supd_restore`?
So something like this?
```
diff --git a/src/reconcile/rec_write.c b/src/reconcile/rec_write.c
index c63f0ae50..573d11db7 100644
--- a/src/reconcile/rec_write.c
+++ b/src/reconcile/rec_write.c
@@ -1904,7 +1904,9 @@ copy_image:
      * If re-instantiating this page in memory (either because eviction wants to, or because we
      * skipped updates to build the disk image), save a copy of the disk image.
      */
-    if (F_ISSET(r, WT_REC_SCRUB) || multi->supd_restore)
+    if (F_ISSET(r, WT_REC_SCRUB)
+       multi->supd_restore = true;
+    if (multi->supd_restore)
         WT_RET(__wt_memdup(session, chunk->image.data, chunk->image.size, &multi->disk_image));
```

Reply:
I think whether scrub needs to be restored still depends on whether there are dirty update chains after reconciliation. If everything is clean after reconciliation, we only need to load the image to memory and no need to restore that. So I don't think we need to handle scrub specially here.

Reply:
If the scrub eviction results to unclean reconciliation, multi->supd_restore must have been set to true already in __rec_supd_move so no need for us to handle it again here.

Reply:
Please create a follow on ticket. It's not obvious to me that WT_REC_SCRUB should free updates that were flushed out to the history store. The goal of WT_REC_SCRUB is to keep content in cache, but minimize how much work needs to be done to flush back dirty data moving forward.

Reply:
Created WT-5947